README.md
# UArmCalibrationTool
GUI uArm Calibration Tool, Build with PyQt4


HISTORY.md
## [1.2.2] - 2016-06-12
### Update - Alex Tan

### Fix
- detect system language.
- catch exit event

## [1.2.1] - 2016-06-12
### Update - Alex Tan

### Fix
- UnkwonFirmwareException Show messagebox


## [1.2] - 2016-06-12
### Update - Alex Tan

### Fix
- Support pyuarm v1.3.x

## [1.1] - 2016-05-12
### Create - Alex Tan

### Changes
- Support Chinese Language
- Move UI files to ui folder

### TO-DO
1. Support Chinese Language
2. Support Firmware update.
3. Support Firmware version online checking

## [1.0] - 2016-05-12
### Create - Alex Tan

### Changes
- uArm Calibration Simple Tool
- 1. Linear Calibration
- 2. Manual Calibration
- 3. Stretch Calibration

### TO-DO
1. Support Chinese Language
2. Support Firmware update.
3. Support Firmware version online checking


resources/languages.ts
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE TS>
<TS version="2.1" language="zh_CN" sourcelanguage="en_US">
<context>
    <name>MainWindow</name>
    <message>
        <location filename="../main_window.py" line="59"/>
        <source>Waiting For Next Action</source>
        <translation>等待下一步操作</translation>
    </message>
    <message>
        <location filename="../main_window.py" line="85"/>
        <source>Trying to Reconnect uArm</source>
        <translation>尝试重新连接uArm</translation>
    </message>
    <message>
        <location filename="../main_window.py" line="102"/>
        <source>Connect to Port: </source>
        <translation>连接到端口：</translation>
    </message>
    <message>
        <location filename="../main_window.py" line="104"/>
        <source>Please Connect uArm</source>
        <translation>请连接uArm</translation>
    </message>
    <message>
        <location filename="../main_window.py" line="130"/>
        <source>Calibration is already start!</source>
        <translation>校正已经开始了！</translation>
    </message>
    <message>
        <location filename="../main_window.py" line="132"/>
        <source>uArm is disconnected, Please check!</source>
        <translation>uArm已经断开了连接，请检查后再试！</translation>
    </message>
    <message>
        <location filename="../main_window.py" line="134"/>
        <source>Please Connect uArm First</source>
        <translation>请先连接uArm</translation>
    </message>
    <message>
        <location filename="../main_window.py" line="173"/>
        <source>Calibrating Linear Offset</source>
        <translation>线性校正偏差</translation>
    </message>
    <message>
        <location filename="../main_window.py" line="177"/>
        <source>Linear Offset Calibration Completed</source>
        <translation>线性校正完成</translation>
    </message>
    <message>
        <location filename="../main_window.py" line="192"/>
        <source>Manual Calibration Start</source>
        <translation>手动校正开始</translation>
    </message>
    <message>
        <location filename="../main_window.py" line="196"/>
        <source>Manual Calibration Completed</source>
        <translation>手动校正完成</translation>
    </message>
    <message>
        <location filename="../main_window.py" line="200"/>
        <source>Please Complete Linear Calibration First</source>
        <translation>请先完成线性校正</translation>
    </message>
    <message>
        <location filename="../main_window.py" line="218"/>
        <source>Stretch Offset Calibration Start</source>
        <translation>长度偏差校正开始</translation>
    </message>
    <message>
        <location filename="../main_window.py" line="223"/>
        <source>Stretch Offset Calibration Completed</source>
        <translation>长度偏差校正完成</translation>
    </message>
    <message>
        <location filename="../main_window.py" line="226"/>
        <source>Please Complete Manual Calibration First</source>
        <translation>请先完成手动校正</translation>
    </message>
</context>
<context>
    <name>loadingWindow</name>
    <message>
        <location filename="../ui/loading_dialog_ui.py" line="37"/>
        <source>Dialog</source>
        <translation>通知</translation>
    </message>
    <message>
        <location filename="../ui/loading_dialog_ui.py" line="38"/>
        <source>Loading, Please Wait...</source>
        <translation>加载中，请稍候...</translation>
    </message>
</context>
<context>
    <name>mainWindow</name>
    <message>
        <location filename="../ui/main_window_ui.py" line="288"/>
        <source>uArm Calibration Tool</source>
        <translation>uArm校正工具</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="289"/>
        <source>Wating For Connect</source>
        <translation>正在等待连接</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="290"/>
        <source>Port:</source>
        <translation>端口：</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="319"/>
        <source>N/A</source>
        <translation>N/A</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="292"/>
        <source>Firmware Version:</source>
        <translation>固件版本：</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="294"/>
        <source>uArm SN:</source>
        <translation>uArm序列号：</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="296"/>
        <source>2. Manual Calibration</source>
        <translation>2. 手动校正</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="297"/>
        <source>Left</source>
        <translation>左</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="299"/>
        <source>Right</source>
        <translation>右</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="301"/>
        <source>Confirm</source>
        <translation>确认</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="314"/>
        <source>1</source>
        <translation>1</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="316"/>
        <source>2</source>
        <translation>2</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="318"/>
        <source>3</source>
        <translation>3</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="311"/>
        <source>4</source>
        <translation>4</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="320"/>
        <source>3. Stretch Calirbation</source>
        <translation>3.长度偏差校正</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="321"/>
        <source>1. Linear Calibration</source>
        <translation>1.线性校正</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="324"/>
        <source>Start</source>
        <translation>开始</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="323"/>
        <source>File</source>
        <translation>文件</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="325"/>
        <source>F4</source>
        <translation>F4</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="326"/>
        <source>Check Update</source>
        <translation>检查更新</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="327"/>
        <source>F9</source>
        <translation>F9</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="328"/>
        <source>Quit</source>
        <translation>退出</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="329"/>
        <source>Refresh</source>
        <translation>刷新</translation>
    </message>
    <message>
        <location filename="../ui/main_window_ui.py" line="330"/>
        <source>F5</source>
        <translation>F5</translation>
    </message>
</context>
</TS>


ui/loading_dialog_ui.py
# -*- coding: utf-8 -*-

# Form implementation generated from reading ui file 'loading_dialog_ui.ui'
#
# Created by: PyQt4 UI code generator 4.11.4
#
# WARNING! All changes made in this file will be lost!

from PyQt4 import QtCore, QtGui

try:
    _fromUtf8 = QtCore.QString.fromUtf8
except AttributeError:
    def _fromUtf8(s):
        return s

try:
    _encoding = QtGui.QApplication.UnicodeUTF8
    def _translate(context, text, disambig):
        return QtGui.QApplication.translate(context, text, disambig, _encoding)
except AttributeError:
    def _translate(context, text, disambig):
        return QtGui.QApplication.translate(context, text, disambig)

class Ui_loadingWindow(object):
    def setupUi(self, loadingWindow):
        loadingWindow.setObjectName(_fromUtf8("loadingWindow"))
        loadingWindow.resize(409, 48)
        self.label = QtGui.QLabel(loadingWindow)
        self.label.setGeometry(QtCore.QRect(110, 20, 241, 16))
        self.label.setObjectName(_fromUtf8("label"))

        self.retranslateUi(loadingWindow)
        QtCore.QMetaObject.connectSlotsByName(loadingWindow)

    def retranslateUi(self, loadingWindow):
        loadingWindow.setWindowTitle(_translate("loadingWindow", "Dialog", None))
        self.label.setText(_translate("loadingWindow", "Loading, Please Wait...", None))



main_window_ui.py
# -*- coding: utf-8 -*-

# Form implementation generated from reading ui file 'main_window_ui.ui'
#
# Created by: PyQt4 UI code generator 4.11.4
#
# WARNING! All changes made in this file will be lost!

from PyQt4 import QtCore, QtGui

try:
    _fromUtf8 = QtCore.QString.fromUtf8
except AttributeError:
    def _fromUtf8(s):
        return s

try:
    _encoding = QtGui.QApplication.UnicodeUTF8
    def _translate(context, text, disambig):
        return QtGui.QApplication.translate(context, text, disambig, _encoding)
except AttributeError:
    def _translate(context, text, disambig):
        return QtGui.QApplication.translate(context, text, disambig)

class Ui_mainWindow(object):
    def setupUi(self, mainWindow):
        mainWindow.setObjectName(_fromUtf8("mainWindow"))
        mainWindow.resize(885, 358)
        self.centralwidget = QtGui.QWidget(mainWindow)
        self.centralwidget.setObjectName(_fromUtf8("centralwidget"))
        self.label_notify = QtGui.QLabel(self.centralwidget)
        self.label_notify.setGeometry(QtCore.QRect(90, 10, 701, 41))
        font = QtGui.QFont()
        font.setPointSize(20)
        self.label_notify.setFont(font)
        self.label_notify.setAlignment(QtCore.Qt.AlignCenter)
        self.label_notify.setObjectName(_fromUtf8("label_notify"))
        self.label_port_text = QtGui.QLabel(self.centralwidget)
        self.label_port_text.setGeometry(QtCore.QRect(40, 70, 59, 16))
        self.label_port_text.setObjectName(_fromUtf8("label_port_text"))
        self.label_port_val = QtGui.QLabel(self.centralwidget)
        self.label_port_val.setGeometry(QtCore.QRect(110, 70, 171, 16))
        self.label_port_val.setObjectName(_fromUtf8("label_port_val"))
        self.label_firmware_version_text = QtGui.QLabel(self.centralwidget)
        self.label_firmware_version_text.setGeometry(QtCore.QRect(350, 70, 121, 16))
        self.label_firmware_version_text.setObjectName(_fromUtf8("label_firmware_version_text"))
        self.label_firmware_version_val = QtGui.QLabel(self.centralwidget)
        self.label_firmware_version_val.setGeometry(QtCore.QRect(500, 70, 59, 16))
        self.label_firmware_version_val.setObjectName(_fromUtf8("label_firmware_version_val"))
        self.label_uarm_sn_text = QtGui.QLabel(self.centralwidget)
        self.label_uarm_sn_text.setGeometry(QtCore.QRect(620, 70, 71, 16))
        self.label_uarm_sn_text.setObjectName(_fromUtf8("label_uarm_sn_text"))
        self.label_uarm_sn_val = QtGui.QLabel(self.centralwidget)
        self.label_uarm_sn_val.setGeometry(QtCore.QRect(708, 70, 131, 20))
        self.label_uarm_sn_val.setObjectName(_fromUtf8("label_uarm_sn_val"))
        self.label_manual_calibration = QtGui.QLabel(self.centralwidget)
        self.label_manual_calibration.setGeometry(QtCore.QRect(340, 110, 141, 51))
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.label_manual_calibration.setFont(font)
        self.label_manual_calibration.setAlignment(QtCore.Qt.AlignCenter)
        self.label_manual_calibration.setObjectName(_fromUtf8("label_manual_calibration"))
        self.verticalLayoutWidget_6 = QtGui.QWidget(self.centralwidget)
        self.verticalLayoutWidget_6.setGeometry(QtCore.QRect(630, 160, 171, 131))
        self.verticalLayoutWidget_6.setObjectName(_fromUtf8("verticalLayoutWidget_6"))
        self.verticalLayout_7 = QtGui.QVBoxLayout(self.verticalLayoutWidget_6)
        self.verticalLayout_7.setSpacing(0)
        self.verticalLayout_7.setObjectName(_fromUtf8("verticalLayout_7"))
        self.horizontalLayout_23 = QtGui.QHBoxLayout()
        self.horizontalLayout_23.setSpacing(0)
        self.horizontalLayout_23.setObjectName(_fromUtf8("horizontalLayout_23"))
        self.label_linear_servo_17 = QtGui.QLabel(self.verticalLayoutWidget_6)
        self.label_linear_servo_17.setAlignment(QtCore.Qt.AlignCenter)
        self.label_linear_servo_17.setObjectName(_fromUtf8("label_linear_servo_17"))
        self.horizontalLayout_23.addWidget(self.label_linear_servo_17)
        self.label_stretch_left_value = QtGui.QLabel(self.verticalLayoutWidget_6)
        self.label_stretch_left_value.setObjectName(_fromUtf8("label_stretch_left_value"))
        self.horizontalLayout_23.addWidget(self.label_stretch_left_value)
        self.f_stretch_left_flag = QtGui.QFrame(self.verticalLayoutWidget_6)
        self.f_stretch_left_flag.setFrameShape(QtGui.QFrame.StyledPanel)
        self.f_stretch_left_flag.setFrameShadow(QtGui.QFrame.Raised)
        self.f_stretch_left_flag.setObjectName(_fromUtf8("f_stretch_left_flag"))
        self.horizontalLayout_23.addWidget(self.f_stretch_left_flag)
        self.verticalLayout_7.addLayout(self.horizontalLayout_23)
        self.horizontalLayout_24 = QtGui.QHBoxLayout()
        self.horizontalLayout_24.setSpacing(0)
        self.horizontalLayout_24.setObjectName(_fromUtf8("horizontalLayout_24"))
        self.label_linear_servo_18 = QtGui.QLabel(self.verticalLayoutWidget_6)
        self.label_linear_servo_18.setAlignment(QtCore.Qt.AlignCenter)
        self.label_linear_servo_18.setObjectName(_fromUtf8("label_linear_servo_18"))
        self.horizontalLayout_24.addWidget(self.label_linear_servo_18)
        self.label_stretch_right_value = QtGui.QLabel(self.verticalLayoutWidget_6)
        self.label_stretch_right_value.setObjectName(_fromUtf8("label_stretch_right_value"))
        self.horizontalLayout_24.addWidget(self.label_stretch_right_value)
        self.f_stretch_right_flag = QtGui.QFrame(self.verticalLayoutWidget_6)
        self.f_stretch_right_flag.setFrameShape(QtGui.QFrame.StyledPanel)
        self.f_stretch_right_flag.setFrameShadow(QtGui.QFrame.Raised)
        self.f_stretch_right_flag.setObjectName(_fromUtf8("f_stretch_right_flag"))
        self.horizontalLayout_24.addWidget(self.f_stretch_right_flag)
        self.verticalLayout_7.addLayout(self.horizontalLayout_24)
        self.button_manual_calibration_confirm = QtGui.QPushButton(self.centralwidget)
        self.button_manual_calibration_confirm.setGeometry(QtCore.QRect(790, 20, 91, 32))
        self.button_manual_calibration_confirm.setObjectName(_fromUtf8("button_manual_calibration_confirm"))
        self.verticalLayoutWidget_2 = QtGui.QWidget(self.centralwidget)
        self.verticalLayoutWidget_2.setGeometry(QtCore.QRect(40, 160, 251, 131))
        self.verticalLayoutWidget_2.setObjectName(_fromUtf8("verticalLayoutWidget_2"))
        self.verticalLayout_5 = QtGui.QVBoxLayout(self.verticalLayoutWidget_2)
        self.verticalLayout_5.setSpacing(0)
        self.verticalLayout_5.setObjectName(_fromUtf8("verticalLayout_5"))
        self.horizontalLayout_16 = QtGui.QHBoxLayout()
        self.horizontalLayout_16.setSpacing(0)
        self.horizontalLayout_16.setObjectName(_fromUtf8("horizontalLayout_16"))
        self.label_linear_servo_10 = QtGui.QLabel(self.verticalLayoutWidget_2)
        self.label_linear_servo_10.setAlignment(QtCore.Qt.AlignCenter)
        self.label_linear_servo_10.setObjectName(_fromUtf8("label_linear_servo_10"))
        self.horizontalLayout_16.addWidget(self.label_linear_servo_10)
        self.label_linear_servo1_intercept = QtGui.QLabel(self.verticalLayoutWidget_2)
        self.label_linear_servo1_intercept.setObjectName(_fromUtf8("label_linear_servo1_intercept"))
        self.horizontalLayout_16.addWidget(self.label_linear_servo1_intercept)
        self.label_linear_servo1_slope = QtGui.QLabel(self.verticalLayoutWidget_2)
        self.label_linear_servo1_slope.setObjectName(_fromUtf8("label_linear_servo1_slope"))
        self.horizontalLayout_16.addWidget(self.label_linear_servo1_slope)
        self.f_linear_servo1_flag = QtGui.QFrame(self.verticalLayoutWidget_2)
        self.f_linear_servo1_flag.setFrameShape(QtGui.QFrame.StyledPanel)
        self.f_linear_servo1_flag.setFrameShadow(QtGui.QFrame.Raised)
        self.f_linear_servo1_flag.setObjectName(_fromUtf8("f_linear_servo1_flag"))
        self.horizontalLayout_16.addWidget(self.f_linear_servo1_flag)
        self.verticalLayout_5.addLayout(self.horizontalLayout_16)
        self.horizontalLayout_17 = QtGui.QHBoxLayout()
        self.horizontalLayout_17.setSpacing(0)
        self.horizontalLayout_17.setObjectName(_fromUtf8("horizontalLayout_17"))
        self.label_linear_servo_11 = QtGui.QLabel(self.verticalLayoutWidget_2)
        self.label_linear_servo_11.setAlignment(QtCore.Qt.AlignCenter)
        self.label_linear_servo_11.setObjectName(_fromUtf8("label_linear_servo_11"))
        self.horizontalLayout_17.addWidget(self.label_linear_servo_11)
        self.label_linear_servo2_intercept = QtGui.QLabel(self.verticalLayoutWidget_2)
        self.label_linear_servo2_intercept.setObjectName(_fromUtf8("label_linear_servo2_intercept"))
        self.horizontalLayout_17.addWidget(self.label_linear_servo2_intercept)
        self.label_linear_servo2_slope = QtGui.QLabel(self.verticalLayoutWidget_2)
        self.label_linear_servo2_slope.setObjectName(_fromUtf8("label_linear_servo2_slope"))
        self.horizontalLayout_17.addWidget(self.label_linear_servo2_slope)
        self.f_linear_servo2_flag = QtGui.QFrame(self.verticalLayoutWidget_2)
        self.f_linear_servo2_flag.setFrameShape(QtGui.QFrame.StyledPanel)
        self.f_linear_servo2_flag.setFrameShadow(QtGui.QFrame.Raised)
        self.f_linear_servo2_flag.setObjectName(_fromUtf8("f_linear_servo2_flag"))
        self.horizontalLayout_17.addWidget(self.f_linear_servo2_flag)
        self.verticalLayout_5.addLayout(self.horizontalLayout_17)
        self.horizontalLayout_18 = QtGui.QHBoxLayout()
        self.horizontalLayout_18.setSpacing(0)
        self.horizontalLayout_18.setObjectName(_fromUtf8("horizontalLayout_18"))
        self.label_linear_servo_12 = QtGui.QLabel(self.verticalLayoutWidget_2)
        self.label_linear_servo_12.setAlignment(QtCore.Qt.AlignCenter)
        self.label_linear_servo_12.setObjectName(_fromUtf8("label_linear_servo_12"))
        self.horizontalLayout_18.addWidget(self.label_linear_servo_12)
        self.label_linear_servo3_intercept = QtGui.QLabel(self.verticalLayoutWidget_2)
        self.label_linear_servo3_intercept.setObjectName(_fromUtf8("label_linear_servo3_intercept"))
        self.horizontalLayout_18.addWidget(self.label_linear_servo3_intercept)
        self.label_linear_servo3_slope = QtGui.QLabel(self.verticalLayoutWidget_2)
        self.label_linear_servo3_slope.setObjectName(_fromUtf8("label_linear_servo3_slope"))
        self.horizontalLayout_18.addWidget(self.label_linear_servo3_slope)
        self.f_linear_servo3_flag = QtGui.QFrame(self.verticalLayoutWidget_2)
        self.f_linear_servo3_flag.setFrameShape(QtGui.QFrame.StyledPanel)
        self.f_linear_servo3_flag.setFrameShadow(QtGui.QFrame.Raised)
        self.f_linear_servo3_flag.setObjectName(_fromUtf8("f_linear_servo3_flag"))
        self.horizontalLayout_18.addWidget(self.f_linear_servo3_flag)
        self.verticalLayout_5.addLayout(self.horizontalLayout_18)
        self.horizontalLayout_19 = QtGui.QHBoxLayout()
        self.horizontalLayout_19.setSpacing(0)
        self.horizontalLayout_19.setObjectName(_fromUtf8("horizontalLayout_19"))
        self.label_linear_servo_13 = QtGui.QLabel(self.verticalLayoutWidget_2)
        self.label_linear_servo_13.setAlignment(QtCore.Qt.AlignCenter)
        self.label_linear_servo_13.setObjectName(_fromUtf8("label_linear_servo_13"))
        self.horizontalLayout_19.addWidget(self.label_linear_servo_13)
        self.label_linear_servo4_intercept = QtGui.QLabel(self.verticalLayoutWidget_2)
        self.label_linear_servo4_intercept.setObjectName(_fromUtf8("label_linear_servo4_intercept"))
        self.horizontalLayout_19.addWidget(self.label_linear_servo4_intercept)
        self.label_linear_servo4_slope = QtGui.QLabel(self.verticalLayoutWidget_2)
        self.label_linear_servo4_slope.setObjectName(_fromUtf8("label_linear_servo4_slope"))
        self.horizontalLayout_19.addWidget(self.label_linear_servo4_slope)
        self.f_linear_servo4_flag = QtGui.QFrame(self.verticalLayoutWidget_2)
        self.f_linear_servo4_flag.setFrameShape(QtGui.QFrame.StyledPanel)
        self.f_linear_servo4_flag.setFrameShadow(QtGui.QFrame.Raised)
        self.f_linear_servo4_flag.setObjectName(_fromUtf8("f_linear_servo4_flag"))
        self.horizontalLayout_19.addWidget(self.f_linear_servo4_flag)
        self.verticalLayout_5.addLayout(self.horizontalLayout_19)
        self.verticalLayoutWidget_4 = QtGui.QWidget(self.centralwidget)
        self.verticalLayoutWidget_4.setGeometry(QtCore.QRect(350, 160, 201, 131))
        self.verticalLayoutWidget_4.setObjectName(_fromUtf8("verticalLayoutWidget_4"))
        self.verticalLayout_6 = QtGui.QVBoxLayout(self.verticalLayoutWidget_4)
        self.verticalLayout_6.setSpacing(0)
        self.verticalLayout_6.setObjectName(_fromUtf8("verticalLayout_6"))
        self.horizontalLayout_20 = QtGui.QHBoxLayout()
        self.horizontalLayout_20.setSpacing(0)
        self.horizontalLayout_20.setObjectName(_fromUtf8("horizontalLayout_20"))
        self.label_linear_servo_14 = QtGui.QLabel(self.verticalLayoutWidget_4)
        self.label_linear_servo_14.setAlignment(QtCore.Qt.AlignCenter)
        self.label_linear_servo_14.setObjectName(_fromUtf8("label_linear_servo_14"))
        self.horizontalLayout_20.addWidget(self.label_linear_servo_14)
        self.label_manual_servo1_value = QtGui.QLabel(self.verticalLayoutWidget_4)
        self.label_manual_servo1_value.setObjectName(_fromUtf8("label_manual_servo1_value"))
        self.horizontalLayout_20.addWidget(self.label_manual_servo1_value)
        self.f_manual_servo1_flag = QtGui.QFrame(self.verticalLayoutWidget_4)
        self.f_manual_servo1_flag.setFrameShape(QtGui.QFrame.StyledPanel)
        self.f_manual_servo1_flag.setFrameShadow(QtGui.QFrame.Raised)
        self.f_manual_servo1_flag.setObjectName(_fromUtf8("f_manual_servo1_flag"))
        self.horizontalLayout_20.addWidget(self.f_manual_servo1_flag)
        self.verticalLayout_6.addLayout(self.horizontalLayout_20)
        self.horizontalLayout_21 = QtGui.QHBoxLayout()
        self.horizontalLayout_21.setSpacing(0)
        self.horizontalLayout_21.setObjectName(_fromUtf8("horizontalLayout_21"))
        self.label_linear_servo_15 = QtGui.QLabel(self.verticalLayoutWidget_4)
        self.label_linear_servo_15.setAlignment(QtCore.Qt.AlignCenter)
        self.label_linear_servo_15.setObjectName(_fromUtf8("label_linear_servo_15"))
        self.horizontalLayout_21.addWidget(self.label_linear_servo_15)
        self.label_manual_servo2_value = QtGui.QLabel(self.verticalLayoutWidget_4)
        self.label_manual_servo2_value.setObjectName(_fromUtf8("label_manual_servo2_value"))
        self.horizontalLayout_21.addWidget(self.label_manual_servo2_value)
        self.f_manual_servo2_flag = QtGui.QFrame(self.verticalLayoutWidget_4)
        self.f_manual_servo2_flag.setFrameShape(QtGui.QFrame.StyledPanel)
        self.f_manual_servo2_flag.setFrameShadow(QtGui.QFrame.Raised)
        self.f_manual_servo2_flag.setObjectName(_fromUtf8("f_manual_servo2_flag"))
        self.horizontalLayout_21.addWidget(self.f_manual_servo2_flag)
        self.verticalLayout_6.addLayout(self.horizontalLayout_21)
        self.horizontalLayout_22 = QtGui.QHBoxLayout()
        self.horizontalLayout_22.setSpacing(0)
        self.horizontalLayout_22.setObjectName(_fromUtf8("horizontalLayout_22"))
        self.label_linear_servo_16 = QtGui.QLabel(self.verticalLayoutWidget_4)
        self.label_linear_servo_16.setAlignment(QtCore.Qt.AlignCenter)
        self.label_linear_servo_16.setObjectName(_fromUtf8("label_linear_servo_16"))
        self.horizontalLayout_22.addWidget(self.label_linear_servo_16)
        self.label_manual_servo3_value = QtGui.QLabel(self.verticalLayoutWidget_4)
        self.label_manual_servo3_value.setObjectName(_fromUtf8("label_manual_servo3_value"))
        self.horizontalLayout_22.addWidget(self.label_manual_servo3_value)
        self.f_manual_servo3_flag = QtGui.QFrame(self.verticalLayoutWidget_4)
        self.f_manual_servo3_flag.setFrameShape(QtGui.QFrame.StyledPanel)
        self.f_manual_servo3_flag.setFrameShadow(QtGui.QFrame.Raised)
        self.f_manual_servo3_flag.setObjectName(_fromUtf8("f_manual_servo3_flag"))
        self.horizontalLayout_22.addWidget(self.f_manual_servo3_flag)
        self.verticalLayout_6.addLayout(self.horizontalLayout_22)
        self.label_stretch_calirbation = QtGui.QLabel(self.centralwidget)
        self.label_stretch_calirbation.setGeometry(QtCore.QRect(620, 110, 141, 51))
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.label_stretch_calirbation.setFont(font)
        self.label_stretch_calirbation.setAlignment(QtCore.Qt.AlignCenter)
        self.label_stretch_calirbation.setObjectName(_fromUtf8("label_stretch_calirbation"))
        self.label_linear_title = QtGui.QLabel(self.centralwidget)
        self.label_linear_title.setGeometry(QtCore.QRect(40, 110, 141, 51))
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.label_linear_title.setFont(font)
        self.label_linear_title.setAlignment(QtCore.Qt.AlignCenter)
        self.label_linear_title.setObjectName(_fromUtf8("label_linear_title"))
        self.button_calibration_start = QtGui.QPushButton(self.centralwidget)
        self.button_calibration_start.setGeometry(QtCore.QRect(10, 20, 71, 32))
        self.button_calibration_start.setObjectName(_fromUtf8("button_calibration_start"))
        mainWindow.setCentralWidget(self.centralwidget)
        self.menubar = QtGui.QMenuBar(mainWindow)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 885, 22))
        self.menubar.setObjectName(_fromUtf8("menubar"))
        self.menuFile = QtGui.QMenu(self.menubar)
        self.menuFile.setObjectName(_fromUtf8("menuFile"))
        mainWindow.setMenuBar(self.menubar)
        self.statusbar = QtGui.QStatusBar(mainWindow)
        self.statusbar.setObjectName(_fromUtf8("statusbar"))
        mainWindow.setStatusBar(self.statusbar)
        self.action_start = QtGui.QAction(mainWindow)
        self.action_start.setObjectName(_fromUtf8("action_start"))
        self.action_check_update = QtGui.QAction(mainWindow)
        self.action_check_update.setObjectName(_fromUtf8("action_check_update"))
        self.action_quit = QtGui.QAction(mainWindow)
        self.action_quit.setObjectName(_fromUtf8("action_quit"))
        self.action_fresh = QtGui.QAction(mainWindow)
        self.action_fresh.setObjectName(_fromUtf8("action_fresh"))
        self.menuFile.addAction(self.action_start)
        self.menuFile.addAction(self.action_fresh)
        self.menuFile.addAction(self.action_check_update)
        self.menuFile.addAction(self.action_quit)
        self.menubar.addAction(self.menuFile.menuAction())

        self.retranslateUi(mainWindow)
        QtCore.QMetaObject.connectSlotsByName(mainWindow)

    def retranslateUi(self, mainWindow):
        mainWindow.setWindowTitle(_translate("mainWindow", "uArm Calibration Tool", None))
        self.label_notify.setText(_translate("mainWindow", "Wating For Connect", None))
        self.label_port_text.setText(_translate("mainWindow", "Port:", None))
        self.label_port_val.setText(_translate("mainWindow", "N/A", None))
        self.label_firmware_version_text.setText(_translate("mainWindow", "Firmware Version:", None))
        self.label_firmware_version_val.setText(_translate("mainWindow", "N/A", None))
        self.label_uarm_sn_text.setText(_translate("mainWindow", "uArm SN:", None))
        self.label_uarm_sn_val.setText(_translate("mainWindow", "N/A", None))
        self.label_manual_calibration.setText(_translate("mainWindow", "2. Manual Calibration", None))
        self.label_linear_servo_17.setText(_translate("mainWindow", "Left", None))
        self.label_stretch_left_value.setText(_translate("mainWindow", "N/A", None))
        self.label_linear_servo_18.setText(_translate("mainWindow", "Right", None))
        self.label_stretch_right_value.setText(_translate("mainWindow", "N/A", None))
        self.button_manual_calibration_confirm.setText(_translate("mainWindow", "Confirm", None))
        self.label_linear_servo_10.setText(_translate("mainWindow", "1", None))
        self.label_linear_servo1_intercept.setText(_translate("mainWindow", "N/A", None))
        self.label_linear_servo1_slope.setText(_translate("mainWindow", "N/A", None))
        self.label_linear_servo_11.setText(_translate("mainWindow", "2", None))
        self.label_linear_servo2_intercept.setText(_translate("mainWindow", "N/A", None))
        self.label_linear_servo2_slope.setText(_translate("mainWindow", "N/A", None))
        self.label_linear_servo_12.setText(_translate("mainWindow", "3", None))
        self.label_linear_servo3_intercept.setText(_translate("mainWindow", "N/A", None))
        self.label_linear_servo3_slope.setText(_translate("mainWindow", "N/A", None))
        self.label_linear_servo_13.setText(_translate("mainWindow", "4", None))
        self.label_linear_servo4_intercept.setText(_translate("mainWindow", "N/A", None))
        self.label_linear_servo4_slope.setText(_translate("mainWindow", "N/A", None))
        self.label_linear_servo_14.setText(_translate("mainWindow", "1", None))
        self.label_manual_servo1_value.setText(_translate("mainWindow", "N/A", None))
        self.label_linear_servo_15.setText(_translate("mainWindow", "2", None))
        self.label_manual_servo2_value.setText(_translate("mainWindow", "N/A", None))
        self.label_linear_servo_16.setText(_translate("mainWindow", "3", None))
        self.label_manual_servo3_value.setText(_translate("mainWindow", "N/A", None))
        self.label_stretch_calirbation.setText(_translate("mainWindow", "3. Stretch Calirbation", None))
        self.label_linear_title.setText(_translate("mainWindow", "1. Linear Calibration", None))
        self.button_calibration_start.setText(_translate("mainWindow", "Start", None))
        self.menuFile.setTitle(_translate("mainWindow", "File", None))
        self.action_start.setText(_translate("mainWindow", "Start", None))
        self.action_start.setShortcut(_translate("mainWindow", "F4", None))
        self.action_check_update.setText(_translate("mainWindow", "Check Update", None))
        self.action_check_update.setShortcut(_translate("mainWindow", "F9", None))
        self.action_quit.setText(_translate("mainWindow", "Quit", None))
        self.action_fresh.setText(_translate("mainWindow", "Refresh", None))
        self.action_fresh.setShortcut(_translate("mainWindow", "F5", None))

